### PR TITLE
Sns icons

### PR DESCRIPTION
**Pull request by spioupiou**
-- I used font awesome for the SNS icons + download icon. It was quite painful because those icons are transparent and I don't really like that, so I added white backgrounds behind each icon as a workaround 🤣  It seems that fontawesome 6 provide new options to change the transparency into a solid color, but we are using an old version of fontawesome which does not have such option 😭 
-- Only the download button is working right now, I'll work on the SNS sharing function later tonight. I think the PR is quite big already so I wanted to open it now.
-- When the user clicks on the download button, a png is automatically saved on his/her local machine. The png is also saved in our app, under public/uploads (yaaay!)

Plan for the future is to 
- either dynamically change the meta tags with the picture saved in our assets
https://stackoverflow.com/questions/66173915/javascrpt-to-change-meta-tag-contents
- OR insert the screenshot in the page (as a display none) and follow the below tutorial for sharing elements in a tweet.
https://www.webdesignerdepot.com/2019/07/how-to-code-a-click-to-tweet-button/ 

I'll try the dynamic changing of our meta tags first and see how it goes :D

Screen Shot 2022-03-07 at 11 33 02

### CODE REVIEW

**Comment by spioupiou** [comment]
New commit because the screenshots did not show all the containers of the page (there were two unnecessary closing divs in results/show and partial _result_chart that caused the vote tally to not be included in the result container) + the public/uploads new folder needs to have some files inside, otherwise git ignores it 😄

File: app/views/results/_result_chart.html.erb [hunk 1]
```diff
@@ -9,7 +9,5 @@
         <%= render  "result_who_voted", instance: @instance, player: player %>
       </div>
     </div>
-  </div>     
-    <% end %>
-</div>
```
Comment (spioupiou) [inline app/views/results/_result_chart.html.erb:14]: extra unnecessary div

File: app/views/results/show.html.erb [hunk 2]
```diff
@@ -6,52 +6,56 @@
...
-                  <%# needed to get last_instance_id for creation of new instance %>
-                  <%= hidden_field_tag 'last_instance_id', @instance.id %>
-                  <%= f.submit "Exit Game", class: "btn btn-primary" %>
-                </div>
-              <% end %>
-            <% end %>
 
-            <%= render "share_modal" %>
-          </div>
+  <!-- This shouldn't get refreshed -->
```
Comment (spioupiou) [inline app/views/results/show.html.erb:32]: New div for SNS buttons and continue button so that they are not affected by AJAX refreshing

**Comment by carlnoval** [comment]
The share buttons look fab! Played with local, alls fine, didn't also see anything else to add/correct. Will merge.

**Pull request merged**